### PR TITLE
🌱 Don't write hardware details into statusAnnotation

### PR DESCRIPTION
**Pull request by fmuyassarov**
**What this PR does / why we need it**:
Don't write hardware details into `statusAnnotation`, because we don't need to move
inspection data to the target cluster via annotation anymore but do it instead via
hardwareData CRD, which is implemented in https://github.com/metal3-io/baremetal-operator/pull/1099.
Partially fixes https://github.com/metal3-io/cluster-api-provider-metal3/issues/266

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]

File: baremetal/metal3machine_manager_test.go
Comment (kashifest) [inline baremetal/metal3machine_manager_test.go:754]: This is a bit unclear to me , why are you unmarshalling an empty object ?

**Comment by fmuyassarov** [comment]
This PR is ready, adding hold just to not merge it before getting green light on https://github.com/metal3-io/baremetal-operator/pull/1099
/hold

**Comment by fmuyassarov** [comment]
/hold cancel
@kashifest this should be good to go know as BMO PR got merged.

**Comment by fmuyassarov** [comment]
Just leaving a note here @kashifest. Feature wise we have tested it with corresponding BMO, dev-env patches. E2E ubuntu test is failing here due to a common issue, not related to this patch. However, we should run E2E at least once. Not to block the feature landing and to avoid introducing a potential bug related to this feature, let's merge it and once E2E main is fixed we can verify the test with these additions.
Just a heads up, integration test is already checking the functionality of HardwareData. The only difference between E2E and integration tests is that we run some extra tests like node reuse etc during E2E. But since the feature is mainly concerned about pivoting, which is tested during integration, we should be safe to assume that patch is okay to land.

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**
